pkg/skaffold/util/util.go

```diff
@@ -287,6 +287,20 @@ func IsDir(path string) bool {
 	return (err == nil || !os.IsNotExist(err)) && info.IsDir()
 }
 
+// IsEmptyDir returns true for empty directories otherwise false
+func IsEmptyDir(path string) bool {
+	d, err := os.Open(path)
+	if err != nil {
+		return false
+	}
+	defer d.Close()
+	ents, err := d.ReadDir(0)
```

Comment:
According to the [docs](https://pkg.go.dev/os#File.ReadDir), `n ≤ 0` will cause all of the entries to be read.  Setting `n = 1` will ensure that we only read the first entry.

Reply:
You're right but setting it to 1 will also return error for empty directories so it was failing. I think the documentation is not too clear:
```
If n <= 0, ReadDir returns all the DirEntry records remaining in the directory. When it succeeds, it returns a nil error (not io.EOF). 
```

You can test it yourself really quick. This is the output when I set `n = 1`:
```
=== RUN   TestIsEmptyDir
    util_test.go:457: bool differ (-got, +want):   bool(
        - 	false,
        + 	true,
          )
--- FAIL: TestIsEmptyDir (0.00s)
```

Reply:
I'm sure I tested this but I deleted my test case.  Trying again, it fails just like you say 🤦 

Easy fix though:
```
	if _, err := d.ReadDir(1); err == io.EOF {
		return true
	}
	return false
```

Reply:
I like this one, thanks! :+1: